Cleanup PDF output

nashif opened the pull request:
- doc: cleanup for PDF
- doc: move kconfig to indices and table
- doc: print tilte of Kconfig on top

Also fixes #6777

nashif committed:
doc: cleanup for PDF

Cleanup docs for PDFs hiding text that is relevant only for online docs.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
doc: move kconfig to indices and table

Right now the kconfig index is randomly placed in the guides. This is a
table similar to other indeces we have and should be considered as part
of an appendix rather than place it in the middle of the documentation.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
doc: print tilte of Kconfig on top

Add the symbol name on top and in the page title. This will enhance
search results and adds a missing page header that is needed.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

gmarull commented on doc/index.rst:
```diff
@@ -126,6 +128,8 @@ licensing, as described in :ref:`Zephyr_Licensing`.
    Indices and Tables
    ******************
```
should likely be excluded, too

nashif replied:
now we have an index for tex and removing lots of :only: usage in main index

gmarull commented on doc/_scripts/gen_kconfig_rest.py:
```diff
@@ -528,9 +528,12 @@ def sym_header_rst(sym):
+           f".. title:: CONFIG_{sym.name}\n\n" \
            f".. kconfig:: CONFIG_{sym.name}\n\n" \
+           f"CONFIG_{sym.name}\n" \
+           f"{'#'*l}\n\n" \
```
nit: `'#' * l`

nashif committed:
doc: start include README after 'getting started'

Otherwise we end up with a self-reference to the same page we are in.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
doc: CI badge should point to main instead of master

Point to the new branch to get the correct status.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
doc: create an index for PDF files

Most of the content in the default index is HTML only, so create an
index just for PDF and keep it clean.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
doc: add a comment about copyright notices

Add a link to best practices about copyright notices and when to and
when not to change them or add them.

Fixes #6777

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

gmarull commented on doc/index-tex.rst:
```diff
@@ -0,0 +1,24 @@
+   application/index.rst
+   reference/index.rst
+   guides/index.rst
+   security/index.rst
```
I'm not a fan of having two index files, we'll need to maintain 2 duplicate toctrees this way. I also don't see why we can't have the Indices and Tables chapter in the latex output.

nashif replied:
The content is minimal, it is no way the same we maintain in the html variant. If you find a better way to have role based TOC, then I am all ears.

The indices are created automatically, no need to add them. In the html index we are just adding a link to the reference.

cfriedt reviewed (approved):
LGTM
